core/store/migrate/migrations/0259_add_workflow_secrets.sql

```diff
@@ -0,0 +1,40 @@
+-- +goose Up
+-- +goose StatementBegin
+-- Create the workflow_artifacts table
+CREATE TABLE workflow_secrets (
+    id SERIAL PRIMARY KEY,
+    secrets_url TEXT UNIQUE, -- base64 encoded URL
```

Comment:
See my comment about whether this can just be stored in plaintext

As an aside, one thing that jump out here is that different URLs can point to the same resource, so we may need to normalize the URLs at some point; I think it's fine to leave for now though.